zcash_client_sqlite/src/wallet.rs

```diff
@@ -1461,11 +1460,12 @@ pub(crate) fn put_block(
 }
 
 /// Inserts information about a mined transaction that was observed to
-/// contain a note related to this wallet into the database.
-pub(crate) fn put_tx_meta<N>(
+/// contain a note or utxo related to this wallet into the database.
+pub(crate) fn put_tx_meta(
     conn: &rusqlite::Connection,
-    tx: &WalletTx<N>,
     height: BlockHeight,
+    tx_index: Option<u32>,
```

Comment:
This is a technically incorrect change: mined transactions always have a `tx_index`, so it is a type error for `height` to be set and `tx_index` to be `None`.

`tx_index` is exposed via the `v_transactions` view, so it is part of the public API. This bugfix is therefore incomplete: it changes the `v_transactions` bug from "never returns transparent transactions" to "returns transparent transactions with an incorrect `tx_index`".

However, we have a limitation here for UTXOs in that `tx_index` is not needed to create spends, so when transparent receives were added, that field was never passed across from `lightwalletd`:
https://github.com/zcash/librustzcash/blob/ba8cb2eff5afdc40d3f4ccaf58443ae3d0a9235c/zcash_client_backend/proto/service.proto#L144-L151

(We only happen to know `height` because it is necessary to determine whether a UTXO has sufficient confirmations, but we don't actually use it for that because we zero-conf spend all UTXOs, so we're just _very_ lucky in that regard.)

The correct fix would be to:
- Change `lightwalletd` to provide `tx_index`.
- Change the mobile SDKs to pass `tx_index` across the FFI.
- Change `WalletTransparentOutput` to hold `tx_index`.
- Pass it into `put_tx_meta`.

That set of changes is too complex to fit into the mobile SDK 2.0.0 release window. So we now need to ask ourselves several questions:
- Is `v_transactions.tx_index` used _anywhere_ that we support? Main places to look are the Android and iOS SDKs.
- Will it cause any non-transient problems for us to expose the wrong value of `tx_index` via a public API?
- Will it cause any non-transient problems for us to store `(height, NULL)` in the wallet DB, in addition to `(NULL, NULL)` and `(height, tx_index)`?